sympy/matrices/common.py

```diff
@@ -673,16 +674,19 @@ def diag(kls, *args, **kwargs):
         ========
 
         >>> from sympy.matrices import Matrix
-        >>> Matrix.diag(1, 2, 3)
-        Matrix([
-        [1, 0, 0],
-        [0, 2, 0],
-        [0, 0, 3]])
-        >>> Matrix.diag([1, 2, 3])
+        >>> diag = Matrix.diag
+
+        All diagonal elements can be given individually
```

Comment:
@siefkenj or @asmeurer, can you tell me why we have two routines that behave differently? From `sympy import *` state, if you use `diag` you get the version that comes from `dense` but if you use `Matrix.diag` you get this one. And this one unpacks a list whereas the former converts lists to Matrices. So in the former you would insert a column down from the diagonal while the in the latter you would get a diagonal of elements: 
```
[1, 0]     [1, 0]
[0. 1] vs [1, 0]
```

Reply:
I don't know why this is (other than it's the way it's been for a long time...). I think that they should have the same behavior, which is the one that `Matrix.diag` has. It's also weird that the docstring is repeated.

Reply:
During the renaming of mutable to dense and the cleanup and separation of matrices to common, the two version persisted and then during the cleanup you added unpacking to one and normalization to the other. So the public facing `diag` does not unpack and converts lists to matrices. I think that makes sense. The codebase also uses it this way when it calls `diag(*foo)`.
I won't break anything but I will get rid of the routine from `dense`.